src/qibo/models/circuit.py

```diff
@@ -203,6 +204,30 @@ def __init__(
                 )
             self._distributed_init(nqubits, accelerators)
 
+    def _parse(self, qubit_spec, wire_names):
```

Comment:
I'd put this out of the class (it does not depend on self), and give it a more telling name (something like `_extract_qubits` or `_determine_qubits`, or similar)

Reply:
[41f5eff](https://github.com/qiboteam/qibo/pull/1500/commits/41f5eff1cc44ad8df1c6c9485eda96d638b4f557)